anvil/plot.py

```diff
@@ -0,0 +1,317 @@
+from math import ceil, floor, log10, fabs
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from matplotlib.ticker import MaxNLocator
+from scipy.signal import correlate
+import torch
+from tqdm import tqdm
+
+from reportengine.table import table
+from reportengine.figure import figure
+
+
+def print_to_precision(val, err):
```

Comment:
what exactly is the problem with fstrings and using the general format?

i.e:
f"{val:.2g}"

Reply:
I think reportengine also does it's own formatting of numbers if used in tables (when tables were printed in reports) was there a problem with that?

Reply:
The precision being dictated by the magnitude of the uncertainty. I may be wrong but I couldn't see how to do that with fstrings etc.

Reply:
oh I see, I think for now I just wouldn't bother formatting the numbers and then for final results worry about this, seems needlessly complicated for production

Reply:
like right now I don't think we're bothered if we type 1e5 +/- 1e-2 it just means we basically have no error